contracts/CurrencyNetwork.sol

```diff
@@ -515,62 +611,70 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
         TrustlineRequest storage trustlineRequest = requestedTrustlineUpdates[uniqueIdentifier(_a, _b)];
         trustlineRequest.creditlineGiven = _trustlineRequest.creditlineGiven;
         trustlineRequest.creditlineReceived = _trustlineRequest.creditlineReceived;
+        if (!customInterests) {
+            trustlineRequest.interestRateGiven = defaultInterests;
+            trustlineRequest.interestRateReceived = defaultInterests;
+        } else {
+            trustlineRequest.interestRateGiven = _trustlineRequest.interestRateGiven;
+            trustlineRequest.interestRateReceived = _trustlineRequest.interestRateReceived;
+        }
         trustlineRequest.initiator = _trustlineRequest.initiator;
     }
 
-    function _updateCreditline(address _creditor, address _debtor, uint32 _value) internal returns (bool success) {
-        Account memory account = _loadAccount(_creditor, _debtor);
-
-        addToUsersAndFriends(_creditor, _debtor);
-        account.creditlineGiven = _value;
-
-        _storeAccount(_creditor, _debtor, account);
-        emit CreditlineUpdate(_creditor, _debtor, _value);
-        success = true;
-    }
-
+    // in this function, it is assumed _creditor is the initator of the trustline update (see _requestTrustlineUpdate())
     function _updateTrustline(
         address _creditor,
         address _debtor,
-        uint32 _creditlineGiven,
-        uint32 _creditlineReceived
+        uint128 _creditlineGiven,
+        uint128 _creditlineReceived,
+        int16 _interestRateGiven,
+        int16 _interestRateReceived
     )
         internal
         returns (bool success)
     {
         Account memory account = _loadAccount(_creditor, _debtor);
 
-        // reduce of creditline is always possible
-        if (_creditlineGiven <= account.creditlineGiven && _creditlineReceived <= account.creditlineReceived) {
+        // reduce of creditlines and interests is always possible
+        if (_creditlineGiven <= account.creditlineGiven && _creditlineReceived <= account.creditlineReceived && _interestRateGiven == account.interestRateGiven && _interestRateReceived == account.interestRateReceived) {
```

Comment:
I am not sure, since interests can be negative and it is unclear to me what people would achieve with it. But I agree that it makes sense to reduce the interestRateGiven (not received).

Reply:
Yes.
Negative interests will be only globally set, so can not be changed by the user. I discussed that with Heiko.

Reply:
have to change it.